src/EventListener/LogRevisionsListener.php

```diff
@@ -389,6 +426,49 @@ private function getInsertRevisionSQL(ClassMetadata $class): string
         return $this->insertRevisionSQL[$class->name];
     }
 
+    /**
+     * @param ClassMetadata $class
+     * @param $targetClass
+     * @param $assoc
+     * @param (((bool|string)[]|mixed|string)[]|bool|int|mixed|null|string)[] $assoc
+     *
+     * @return string
+     *
+     * @psalm-param ClassMetadata<object> $targetClass
+     * @psalm-param array{cache?: array, cascade: array<string>, declared?: class-string, fetch: mixed, fieldName: string, id?: bool, inherited?: class-string, indexBy?: string, inversedBy: null|string, isCascadeRemove: bool, isCascadePersist: bool, isCascadeRefresh: bool, isCascadeMerge: bool, isCascadeDetach: bool, isOnDeleteCascade?: bool, isOwningSide: true, joinColumns?: array<array{name: string, referencedColumnName: string, unique?: bool, quoted?: bool, fieldName?: string, onDelete?: string, columnDefinition?: string, nullable?: bool}>, joinColumnFieldNames?: array<string, string>, joinTable?: array, joinTableColumns?: list<mixed>, mappedBy: null|string, orderBy?: array, originalClass?: class-string, originalField?: string, orphanRemoval?: bool, relationToSourceKeyColumns?: array, relationToTargetKeyColumns?: array, sourceEntity: class-string, sourceToTargetKeyColumns?: array<string, string>, targetEntity: class-string, targetToSourceKeyColumns?: array<string, string>, type: int, unique?: bool} $assoc
```

Comment:
You can use @phpstan-import-type AssociationMapping from ... https://github.com/doctrine/orm/blob/2.13.x/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php#L110
instead on the class level.

And then use it in the phpdoc.